### PR TITLE
expose polynomial:power and cosine_with_restarts:num_cycles params

### PR DESCRIPTION
**Pull request by zetyquickly**
to get_scheduler func, 

* add it to the train_dreambooth.py

### CODE REVIEW

**Comment by zetyquickly** [comment]
@pcuenca PR to an issue https://github.com/huggingface/diffusers/issues/1232

What's done:
* exposed only two parameters power and num_cycles on top of existing get_scheduler func API. it's possible to expose all of the parameters but taking into account that the issue wasn't there before we've noticed it, I assume it would pile up the code
* changed parameter num_cycles name of the cosine scheduler to not overlap with cosine_with_restarts, because they have different meanings
* added new parameters to the dreambooth's args

Haven't found any test script in test folder that tests optimizers.py

File: src/diffusers/optimization.py [hunk 1]
```diff
@@ -121,9 +121,9 @@ def get_cosine_schedule_with_warmup(
             The number of steps for the warmup phase.
         num_training_steps (`int`):
             The total number of training steps.
-        num_cycles (`float`, *optional*, defaults to 0.5):
-            The number of waves in the cosine schedule (the defaults is to just decrease from the max value to 0
-            following a half-cosine).
+        num_periods (`float`, *optional*, defaults to 0.5):
+            The number of periods of the cosine function in a schedule (the defaults is to just decrease from the max
```
Comment (pcuenca) [inline src/diffusers/optimization.py:125]:
```suggestion
            The number of periods of the cosine function in a schedule (the default is to just decrease from the max
```

**Review by pcuenca — APPROVED** [review]
Looks good!

An alternative would be to use `scheduler_kwargs` in `get_scheduler`, but since we only have those two configurable schedulers I think it's good for now.

**Review by patrickvonplaten — APPROVED** [review]
Thanks @zetyquickly

**Pull request merged**